Sources/PostgresNIO/New/Connection State Machine/ExtendedQueryStateMachine.swift

```diff
@@ -323,20 +385,32 @@ struct ExtendedQueryStateMachine {
     }
     
     mutating func emptyQueryResponseReceived() -> Action {
-        guard case .bindCompleteReceived(let queryContext) = self.state else {
-            return self.setAndFireError(.unexpectedBackendMessage(.emptyQueryResponse))
-        }
+        switch self.state {
+        case .bindCompleteReceived(let queryContext):
+            switch queryContext.query {
+            case .unnamed(_, let eventLoopPromise),
+                 .executeStatement(_, let eventLoopPromise):
+                return self.avoidingStateMachineCoW { state -> Action in
+                    state = .emptyQueryResponseReceived
+                    let result = QueryResult(value: .noRows(.emptyResponse), logger: queryContext.logger)
+                    return .succeedQuery(eventLoopPromise, with: result)
+                }
 
-        switch queryContext.query {
-        case .unnamed(_, let eventLoopPromise),
-             .executeStatement(_, let eventLoopPromise):
-            return self.avoidingStateMachineCoW { state -> Action in
-                state = .emptyQueryResponseReceived
-                let result = QueryResult(value: .noRows(.emptyResponse), logger: queryContext.logger)
-                return .succeedQuery(eventLoopPromise, with: result)
+            case .prepareStatement, .simpleQuery:
+                return self.setAndFireError(.unexpectedBackendMessage(.emptyQueryResponse))
             }
-
-        case .prepareStatement(_, _, _, _):
+        case .messagesSent(let queryContext):
+            switch queryContext.query {
+            case .simpleQuery(_, let eventLoopPromise):
+                return self.avoidingStateMachineCoW { state -> Action in
+                    state = .emptyQueryResponseReceived
+                    let result = QueryResult(value: .noRows(.emptyResponse), logger: queryContext.logger)
+                    return .succeedQuery(eventLoopPromise, with: result)
+                }
```

Comment:
If a `EmptyQuery` is received and state is `messageSent`, It means we had a query that was empty, like only contained some comments.